core/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -1102,6 +1106,7 @@ private void sendRequestToChannel(final DiscoveryNode node, final Channel target
     private void internalSendMessage(Channel targetChannel, BytesReference message, NotifyOnceListener<Channel> listener) {
         try {
             sendMessage(targetChannel, message, listener);
+            txMetric.inc(message.length());
```

Comment:
So if I had one concern with this PR, it is here.

Netty (and other nio options) will not throw on `sendMessage`. So just because that method returned without exception, does not mean the data was sent.

The write could be cancelled due to a `ChannelClosedException` (or any other type of `IOException`). If you only want to increment that counter in case of a successful send, you will need to put it in the `ActionListener`.